cairo_zero/kakarot/library.cairo

```diff
@@ -125,8 +129,8 @@ namespace Kakarot {
     func eth_chain_id{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
         chain_id: felt
     ) {
-        let (tx_info) = get_tx_info();
-        let (_, chain_id) = unsigned_div_rem(tx_info.chain_id, Constants.MAX_SAFE_CHAIN_ID);
+        let (chain_id) = Kakarot_chain_id.read();
+        let (_, chain_id) = unsigned_div_rem(chain_id, Constants.MAX_SAFE_CHAIN_ID);
```

Comment:
I think we could keep it as a fail safe to ensure that evm wallets (metamask at least) will work if an operator decide to choose a chain id > 4 bytes

Reply:
who is "an operator" in that scenario?

Reply:
app chains ?
Someone deploying the kakarot contract

Reply:
the % operation was there because we were forced to use the chain_id of the underlying starknet chain, now that this limitaiton is not there, I don't really see a reason to keep it, the operator can set whatever chain_id he wants. If it's about choosing the right chain id that can be in the documentation

Reply:
At first yes, but then there was a new cause which the compatibility with metamask.
If we remove the check, you could then let the operator set a chain id that is not compatible with it.
Do you think it is a problem to let this fail safe ?

Reply:
I do think it's a problem. The operator picks a chain ID X, there is no reason for the actual chain ID used to be X % n when the operator could have chosen X % n directly.

Reply:
Ok let 's remove it then and add it to the documentation